File: app/src/main/java/com/shindra/stationAdapter.java
package com.shindra;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.shindra.ctslibrary.bo.Line;
import com.shindra.ctslibrary.bo.Stop;

import java.text.SimpleDateFormat;
import java.util.ArrayList;

public class stationAdapter extends RecyclerView.Adapter<stationAdapter.stationDiaryViewHolder>{

    private ArrayList<Stop> _listOfStop;
    private Line _line;

    public static class stationDiaryViewHolder extends RecyclerView.ViewHolder{
        public TextView stationNameView;
        public TextView lineNameView;
        public TextView scheduledTimeView;

        public stationDiaryViewHolder(@NonNull View itemView) {
            super(itemView);
            stationNameView = itemView.findViewById(R.id.stationName);
            lineNameView = itemView.findViewById(R.id.tramLine);
            scheduledTimeView = itemView.findViewById(R.id.scheduledTime);
        }

        public void onBind(Line line, Stop stop, int position){
            stationNameView.setText(stop.component1());
            scheduledTimeView.setText(new SimpleDateFormat("HH'h'mm").format(stop.component2()));
            lineNameView.setText(lineNameView.getContext().getString(R.string.ligneText) + line.getName());
            lineNameView.setTextColor(ContextCompat.getColor(lineNameView.getContext(),getLineColorFromLine(line)));
        }
    }

    public stationAdapter(Line line) {
        _line = line;
        _listOfStop = fillStopListFromLine(line);
    }

    @NonNull
    @Override
    public stationDiaryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View stationCardView = LayoutInflater.from(parent.getContext()).inflate(R.layout.cardview_station, parent, false);
        stationAdapter.stationDiaryViewHolder stationDiaryCardVH = new stationAdapter.stationDiaryViewHolder(stationCardView);
        return stationDiaryCardVH;
    }

    @Override
    public void onBindViewHolder(@NonNull stationDiaryViewHolder holder, int position) {

        Stop currentStop = _listOfStop.get(position);
        holder.onBind(_line,currentStop, position);
    }

    @Override
    public int getItemCount() {
        return _listOfStop.size();
    }

    public static int getLineColorFromLine(Line line){

        switch (line.getName())
        {
            case "A":
                return R.color.ligneA;
            case "B":
                return R.color.ligneB;
            case "C":
                return R.color.ligneC;
            case "D":
                return R.color.ligneD;
            case "E":
                return R.color.ligneE;
            case "F":
                return R.color.ligneF;
            default:
                return R.color.black;
        }
    }

    private  ArrayList<Stop> fillStopListFromLine(Line data){
       ArrayList<Stop> tmpList = new ArrayList<Stop>();

        for (Stop stop : data.getStops())
        {
            tmpList.add(stop);
        }
        return tmpList;
    }

}

File: app/src/main/java/com/shindra/stationAdapter.kt
package com.shindra

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.shindra.ctslibrary.bo.Line
import com.shindra.ctslibrary.bo.Stop
import com.shindra.stationAdapter.stationDiaryViewHolder
import java.text.SimpleDateFormat
import java.util.*

class stationAdapter(private val _line: Line) :
    RecyclerView.Adapter<stationDiaryViewHolder>() {
    private val _listOfStop: ArrayList<Stop>

    class stationDiaryViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var stationNameView: TextView
        var lineNameView: TextView
        var scheduledTimeView: TextView
        fun onBind(line: Line, stop: Stop) {
            stationNameView.text = stop.name
            scheduledTimeView.text = SimpleDateFormat("HH'h'mm").format(stop.estimatedArrivalTime)
            lineNameView.text = lineNameView.context.getString(R.string.ligneText) + line.name
            lineNameView.setTextColor(
                ContextCompat.getColor(
                    lineNameView.context,
                    getLineColorFromLine(line)
                )
            )
        }

        init {
            stationNameView = itemView.findViewById(R.id.stationName)
            lineNameView = itemView.findViewById(R.id.tramLine)
            scheduledTimeView = itemView.findViewById(R.id.scheduledTime)
        }
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): stationDiaryViewHolder {
        val stationCardView = LayoutInflater.from(parent.context)
            .inflate(R.layout.cardview_station, parent, false)
        return stationDiaryViewHolder(stationCardView)
    }

    override fun onBindViewHolder(
        holder: stationDiaryViewHolder,
        position: Int
    ) {
        val currentStop = _listOfStop[position]
        holder.onBind(_line, currentStop)
    }

    override fun getItemCount(): Int {
        return _listOfStop.size
    }

    private fun fillStopListFromLine(data: Line): ArrayList<Stop> {
        val tmpList = ArrayList<Stop>()
        for (stop in data.stops!!) {
            tmpList.add(stop)
        }
        return tmpList
    }

    companion object {
        fun getLineColorFromLine(line: Line): Int {
            return when (line.name) {
                "A" -> R.color.ligneA
                "B" -> R.color.ligneB
                "C" -> R.color.ligneC
                "D" -> R.color.ligneD
                "E" -> R.color.ligneE
                "F" -> R.color.ligneF
                else -> R.color.black
            }
        }
    }

    init {
        _listOfStop = fillStopListFromLine(_line)
    }
}